File: Go/443. String Compression.go
package main

import "strconv"

//goland:noinspection GoUnusedFunction
func compress(chars []byte) int {
	length := len(chars)
	index := 0

	for i := 0; i < length; {
		current := chars[i]
		count := 0

		for i < length && chars[i] == current {
			count++
			i++
		}

		chars[index] = current
		index++

		if count > 1 {
			for _, char := range []byte(strconv.Itoa(count)) {
				chars[index] = char
				index++
			}
		}
	}
	return index
}

File: TypeScript/443. String Compression.ts
// noinspection JSUnusedGlobalSymbols
const compress = (chars: string[]): number => {
    const length = chars.length
    let index = 0

    for (let i = 0; i < length;) {
        const currentChar = chars[i]
        let count = 0

        while (i < length && chars[i] === currentChar) {
            count++
            i++
        }

        chars[index++] = currentChar

        if (count > 1) {
            for (const char of count.toString()) {
                chars[index++] = char
            }
        }
    }
    return index
}
